Source/Filters/Trim/YPTimeStampTrimmerView.swift

```diff
@@ -0,0 +1,213 @@
+//
+//  YPTimeStampTrimmerView.swift
+//  YPImagePicker
+//
+//  Created by Zeph Cohen on 1/16/24.
+//  Copyright © 2024 Yummypets. All rights reserved.
+//
+
+import AVFoundation
+import UIKit
+import PryntTrimmerView
+
+public protocol YPTimeStampTrimmerViewDelegate: AnyObject {
+    func positionBarDidStopMoving(_ playerTime: CMTime)
+    func didChangePositionBar(to playerTime: CMTime)
+}
+
+public class YPTimeStampTrimmerView: UIView {
+
+    // MARK: - Properties
+
+    var timeBarLargeCircleColor: UIColor?
+    var timeBarSmallCircleColor: UIColor?
+    var timeStampFont: UIFont?
+    var timeStampColor: UIColor?
+    var trimMaskolor: UIColor?
+
+    let trimmerView = TrimmerView()
+    let timeStampScrollableView = YPTimeStampScrollableView()
+    let rightHandleTimeStamp = UILabel()
+    let leftHandleTimeStamp = UILabel()
+    private(set) var rightHandleTimeStampConstraint: NSLayoutConstraint?
+    private(set) var leftHandleTimeStampConstraint: NSLayoutConstraint?
+    private(set) var trimViewTopAnchor: NSLayoutConstraint?
+    private(set) var isLaidOut = false
+    public var style: TrimmerStyle = .default
+    weak var timeStampTrimmerViewDelegate: YPTimeStampTrimmerViewDelegate?
+
+    public enum Constant {
+        public static let timeStampTrimViewPadding: CGFloat = 30
+    }
+
+    public enum TrimmerStyle {
+        case `default`
+        case trimmerWithTimeStamps
+    }
+
+    public var startTime: CMTime? {
+        trimmerView.startTime
+    }
+
+    public var endTime: CMTime? {
+        trimmerView.endTime
+    }
+
+    public var asset: AVAsset? {
+        trimmerView.asset
+    }
+
+    // MARK: - Init
+
+    public override func layoutSubviews() {
+        if !isLaidOut {
```

Comment:
do we need to call `super.layoutSubviews()` here

Reply:
@MikkoKuivanenRS Updated here: https://github.com/rewardStyle/YPImagePicker/pull/34/commits/9caf06077c0a498c33c515749ceefd695135079e